### PR TITLE
Enable validation/collation v2 protocols

### PR DESCRIPTION
**Pull request by sandreim**
Removes the `network-protocol-staging` feature. This is needed for the first release of async backing capable nodes.
Renames `VStaging` to `V2`
on top of https://github.com/paritytech/polkadot-sdk/pull/1543

### CODE REVIEW

File: polkadot/node/network/protocol/src/peer_set.rs [hunk 1]
```diff
@@ -118,13 +118,6 @@ impl PeerSet {
...
 	pub fn get_main_version(self) -> ProtocolVersion {
-		#[cfg(not(feature = "network-protocol-staging"))]
-		match self {
-			PeerSet::Validation => ValidationVersion::V1.into(),
-			PeerSet::Collation => CollationVersion::V1.into(),
-		}
-
-		#[cfg(feature = "network-protocol-staging")]
 		match self {
 			PeerSet::Validation => ValidationVersion::VStaging.into(),
```
Comment (sandreim) [inline polkadot/node/network/protocol/src/peer_set.rs:122]: VStaging should be renamed to V2

**Comment by sandreim** [comment]
It makes sense to  wait merging https://github.com/paritytech/polkadot-sdk/pull/1543 first.

File: polkadot/node/network/protocol/src/peer_set.rs [hunk 2]
```diff
@@ -233,7 +226,7 @@ pub enum CollationVersion {
 	/// The first version.
 	V1 = 1,
 	/// The staging version.
```
Comment (slumber) [inline polkadot/node/network/protocol/src/peer_set.rs:228]:
```suggestion
	/// The second version.
```

File: polkadot/node/network/protocol/src/request_response/vstaging.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@ use polkadot_primitives::{
 };
 
 use super::{IsRequest, Protocol};
-use crate::vstaging::StatementFilter;
+use crate::v2::StatementFilter;
```
Comment (slumber) [inline polkadot/node/network/protocol/src/request_response/vstaging.rs:27]: This file should be empty

**Review by ordian — APPROVED** [review]
I hope this won't break Rococo and Westend 😅

**Comment by sandreim** [comment]
> I hope this won't break Rococo and Westend 😅

we could do a burn-in to be sure :)

**Review by slumber — APPROVED** [review]

**Pull request merged**